### DISCUSSION TITLE
What is the proper way to use a secret manager with Next.js?

### DISCUSSION THREAD

**Discussion by dangtony98** [vercel/next.js#47139] [Help]
### Summary

Secret managers like AWS Secret Manager or Parameter Store store secrets and provide an SDK for secrets to be fetched back into apps.

There are plenty examples of using a secret manager with a standard Node backend. With AWS SDK, you typically enter `AWS_SECRET_KEY` and `AWS_ACCESS_KEY` to authenticate with the secret manager and then call methods like `.getSecretValue()` to get secret values back.

That said, I'm confused what the standard practice would be for a Next.js app. My initial thought would be to use `getServerSideProps` to connect and fetch secrets back for each page that needs it but somehow I'd like to ideally avoid `getServerSideProps` entirely and transition the entire app to be fully client-side rendered.

Where would I load in `AWS_SECRET_KEY` and `AWS_ACCESS_KEY` and call `.getSecretValue()` in my app?

**Comment by TorrCod** [comment]
hey have you found answer? im quiet loss

**Comment by bharat5604** [comment]
I am also working on NextJS project and trying to load secrets from aws secret manager. but not getting the proper way of invoking the function. tried invoking in layout page, but this will cause memory leak i guess.

**Comment by syedali-intellia** [comment]
any update ?

**Comment by MorelSerge** [comment]
In general, it would be useful to have access to the extension that AWS has made for Lambda, to access secrets in a cost-efficient way:
https://docs.aws.amazon.com/secretsmanager/latest/userguide/retrieving-secrets_lambda.html

This would open up the possibility to centralise secret access management.
